### PR TITLE
refactor(codewhisperer): remove duplicate code

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
#### Auto trigger

https://github.com/aws/aws-toolkit-vscode/assets/96078566/3390f639-f3e3-4ef1-846a-e3d9d41037d9

#### Manual trigger

https://github.com/aws/aws-toolkit-vscode/assets/96078566/b3579ce5-d712-46d1-8e8f-647250fc6dbd

#### Auto trigger

https://github.com/aws/aws-toolkit-vscode/assets/96078566/376492c9-ac77-4eb6-b683-509dd155bf42

#### Manual trigger

https://github.com/aws/aws-toolkit-vscode/assets/96078566/b33a1557-b057-487e-bc05-65982361465f

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/codewhisperer/commands/invokeRecommendation.ts [hunk 1]
```diff
@@ -4,17 +4,12 @@
  */
 
 import * as vscode from 'vscode'
-import { vsCodeState, ConfigurationEntry, GetRecommendationsResponse } from '../models/model'
```
Comment (leigaol) [inline src/codewhisperer/commands/invokeRecommendation.ts:7]: Can approve once it is tested in Cloud9

**Comment by Will-ShaoHua** [comment]
will wait until post re:invent

File: src/codewhisperer/service/recommendationService.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import { DefaultCodeWhispererClient } from '../client/codewhisperer'
+
+export class RecommendationService {
+    static #instance: RecommendationService
+
+    public static get instance() {
+        return (this.#instance ??= new RecommendationService())
+    }
+
+    async generateRecommednation(
```
Comment (zixlin7) [inline src/codewhisperer/service/recommendationService.ts:24]: nit: typo

**Review by zixlin7 — APPROVED** [review]

**Review by justinmk3 — APPROVED** [review]
is this ready?

**Comment by Will-ShaoHua** [comment]
> is this ready?

@justinmk3 yea it's ready but should we wait until next year given it's not a P0 change, considering it's late Dec and we might have lower coverage for CX's issues in the following 1-2 wks?

**Comment by Will-ShaoHua** [comment]
@justinmk3 this one is ready to go thanks

think the failing test is not relevant to the change here?
```
  1) Auth
    at MultiReporters.done (d:\a\aws-toolkit-vscode\aws-toolkit-vscode\node_modules\mocha-multi-reporters\lib\MultiReporters.js:124:16)
       Shared ini files
    at done (d:\a\aws-toolkit-vscode\aws-toolkit-vscode\node_modules\mocha\lib\mocha.js:1007:16)
         does not cache if the credentials file changes:
    at processTicksAndRejections (node:internal/process/task_queues:95:5)

Extension host test runner error Error: 1 tests failed.
      AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
    at d:\a\aws-toolkit-vscode\aws-toolkit-vscode\src\test\testRunner.ts:90:28
+ actual - expected
    at MultiReporters.done (d:\a\aws-toolkit-vscode\aws-toolkit-vscode\node_modules\mocha-multi-reporters\lib\MultiReporters.js:124:16)

    at done (d:\a\aws-toolkit-vscode\aws-toolkit-vscode\node_modules\mocha\lib\mocha.js:1007:16)
  {
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
+   accessKeyId: 'x',
+   secretAccessKey: 'x',
-   accessKeyId: 'y',
-   secretAccessKey: 'y',
    sessionToken: undefined
  }
      + expected - actual

       {
      -  "accessKeyId": "x"
      -  "secretAccessKey": "x"
      +  "accessKeyId": "y"
      +  "secretAccessKey": "y"
         "sessionToken": [undefined]
       }
      
```

**Pull request merged**